lib/dao.js

```diff
@@ -507,6 +507,11 @@ DataAccessObject.upsert = function(data, options, cb) {
     return this.create(data, options, cb);
   }
 
+  var forceId = this.settings.forceId;
+  if (forceId) {
+    return Model.replaceById(id, data, options, cb);
```

Comment:
`upsert` (`patchOrCreate`) needs to call `prototype.patchAttributes`, not `replaceById`.

(Ideally, it should call `patchById`, but that function is not implemented yet.)